CromIAM/src/test/scala/cromiam/webservice/SubmissionSupportSpec.scala

```diff
@@ -0,0 +1,41 @@
+package cromiam.webservice
+
+import akka.event.{LoggingAdapter, NoLogging}
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{Authorization, OAuth2BearerToken, RawHeader}
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import cromwell.util.SampleWdl.HelloWorld
+import org.scalatest.{FlatSpec, Matchers}
+import spray.json.DefaultJsonProtocol._
+import spray.json._
+
+class SubmissionSupportSpec extends FlatSpec with Matchers with ScalatestRouteTest with SubmissionSupport {
+  override val cromwellClient = new MockCromwellClient()
+  override val samClient = new MockSamClient()
+  override val log: LoggingAdapter = NoLogging
+
+  val authorization = Authorization(OAuth2BearerToken("my-token"))
+  val badAuthHeaders: List[HttpHeader] = List(authorization, RawHeader("OIDC_CLAIM_user_id", samClient.unauthorizedUserCollectionStr))
+  val goodAuthHeaders: List[HttpHeader] = List(authorization, RawHeader("OIDC_CLAIM_user_id", samClient.authorizedUserCollectionStr))
+
+  val submitPath: String = "/api/workflows/v1"
+
+  val helloWorldWdl = HelloWorld.workflowSource()
+  val helloWorldInputs = HelloWorld.rawInputs
+
+  val workflowSource = Multipart.FormData.BodyPart("workflowSource", HttpEntity(MediaTypes.`application/json`, helloWorldWdl))
+  val workflowInputs = Multipart.FormData.BodyPart("workflowInputs", HttpEntity(MediaTypes.`application/json`, helloWorldInputs.toJson.toString))
+  val formData = Multipart.FormData(workflowSource, workflowInputs).toEntity()
+
+  "Submit query" should "forward the request to Cromwell for authorized SAM user" in {
+    Post(submitPath).withHeaders(goodAuthHeaders).withEntity(formData) ~> submitRoute ~> check {
+      status shouldEqual StatusCodes.InternalServerError
```

Comment:
TOL: Looking at the mock code I was able to figure out that this is the expected response because the mock Cromwell client always returns 500, but it's not very obvious here IMO that a test for "user is authorized" should return 500.
I think it would be nice if possible to have the mock cromwell client respond differently based on the request (like you did for SAM). It could send back 200 for authorized and maybe throw an exception otherwise.
This would also have the added benefit to fail any test where Cromwell receives a request that wasn't authorized by CromIAM. For instance in the test below, it would make sure that on top of CromIAM responding with 400, it really didn't send the request to Cromwell.

Reply:
Good point. I will try that. Thanks!